web-marketplace/src/legacy-pages/Dashboard/Dashboard.tsx

```diff
@@ -113,37 +173,313 @@ export const Dashboard = () => {
 
   const onNavClick = (sectionName: string): void => {
     const isFormDirty = isDirtyRef.current;
-    const path = `/dashboard/${sectionName.replace(' ', '-')}`;
+    const path = sectionName
+      ? `${dashboardBasePath}/${sectionName.replace(' ', '-')}`
+      : dashboardBasePath;
     if (isFormDirty) {
       setIsWarningModalOpen(path);
     } else {
       navigate(path);
     }
   };
-  // Placeholder function for account selection logic
-  // This should be replaced with actual logic to handle
-  // account selection when Org functionality is implemented
+
+  const personalResolvedAddress = useMemo(() => {
+    const walletAddress = getWalletAddress({ activeAccount, wallet });
+    if (walletAddress) return walletAddress;
+    return privActiveAccount?.email || '';
+  }, [activeAccount, wallet, privActiveAccount?.email]);
+
+  const daos = activeAccount?.daosByAssignmentAccountIdAndDaoAddress?.nodes;
+
+  const organizationDao = useMemo(
+    () => daos?.find(dao => !!dao?.organizationByDaoAddress) ?? undefined,
+    [daos],
+  );
```

Comment:
you could just use `useDaoOrganization` instead of repeating the same logic here